pkg/datarepair/repairer/repairer.go

```diff
@@ -45,23 +45,22 @@ func (r *repairer) Run(ctx context.Context) (err error) {
 	for {
 		select {
 		case <-r.ticker.C: // wait for the next interval to happen
+			seg, err := r.queue.Dequeue()
+			if err != nil {
+				// TODO: only log when err != ErrQueueEmpty
+				zap.L().Error("dequeue", zap.Error(err))
+				continue
+			}
+
+			r.limiter.Go(ctx, func() {
+				err := r.Repair(ctx, &seg)
+				if err != nil {
+					zap.L().Error("Repair failed", zap.Error(err))
+				}
+			})
 		case <-ctx.Done(): // or the repairer is canceled via context
 			return ctx.Err()
 		}
-
-		seg, err := r.queue.Dequeue()
```

Comment:
Why was this part changed? The reason it was this way separated is to make clear what select is waiting on.